### PR TITLE
Modernize for C++11

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

File: SoftwareGuide/Cover/Source/AntialiasFilter.cxx [hunk 1]
```diff
@@ -36,27 +36,27 @@ int main( int argc, char ** argv )
     }
 
 
-  typedef unsigned char        InputPixelType;
-  typedef float                OutputPixelType;
+  using InputPixelType = unsigned char;
+  using OutputPixelType = float;
 
   const   unsigned int        Dimension = 3;
```
Comment (dzenanz) [inline SoftwareGuide/Cover/Source/AntialiasFilter.cxx]: Just one space in front of Dimension. No need to align it any more!

File: SoftwareGuide/Cover/Source/AntialiasFilter.cxx [hunk 2]
```diff
@@ -36,27 +36,27 @@ int main( int argc, char ** argv )
...
                                   InputImageType,
-                                  OutputImageType > FilterType;
+                                  OutputImageType >;
 
   FilterType::Pointer filter = FilterType::New();
 
 
-  const double            maximumRMSError     =  atof( argv[3] );
-  const unsigned int long numberOfIterations  =  atol( argv[4] );
+  const double            maximumRMSError     = atof( argv[3] );
```
Comment (dzenanz) [inline SoftwareGuide/Cover/Source/AntialiasFilter.cxx]: Do we need to align definitions of constants?

File: SoftwareGuide/Cover/Source/BinaryMaskMedianFilter.cxx [hunk 3]
```diff
@@ -36,19 +36,19 @@ int main( int argc, char ** argv )
     }
 
 
-  typedef unsigned char        PixelType;
+  using PixelType = unsigned char;
 
   const   unsigned int        Dimension = 3;
```
Comment (dzenanz) [inline SoftwareGuide/Cover/Source/BinaryMaskMedianFilter.cxx]: Alignment.

File: SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx [hunk 4]
```diff
@@ -16,20 +16,19 @@ int main(int argc, char * argv[] )
     return -1;
   }
 
-  typedef signed short                          InputPixelType;
-  typedef itk::Image< InputPixelType,     3 >   InputImageType;
+  using InputPixelType = signed short;
+  using InputImageType = itk::Image< InputPixelType,     3 >;
```
Comment (dzenanz) [inline SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx]: Just one space in front of 3 is enough.

File: SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx [hunk 5]
```diff
@@ -16,20 +16,19 @@ int main(int argc, char * argv[] )
...
 
 
-  typedef unsigned char                          OutputPixelType;
-  typedef itk::Image< OutputPixelType,     3 >   OutputImageType;
+  using OutputPixelType = unsigned char;
+  using OutputImageType = itk::Image< OutputPixelType,     3 >;
```
Comment (dzenanz) [inline SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx]: One space

File: SoftwareGuide/Cover/Source/DilateFilter.cxx [hunk 6]
```diff
@@ -37,27 +37,26 @@ int main( int argc, char ** argv )
     }
 
 
-  typedef unsigned char        PixelType;
+  using PixelType = unsigned char;
 
   const   unsigned int        Dimension = 3;
```
Comment (dzenanz) [inline SoftwareGuide/Cover/Source/DilateFilter.cxx]: Alignment

File: SoftwareGuide/Cover/Source/ImageReadExtractWriteRGB.cxx [hunk 7]
```diff
@@ -37,23 +37,23 @@ int main( int argc, char ** argv )
     }
 
 
-  typedef unsigned char        PixelComponentType;
+  using PixelComponentType = unsigned char;
 
-  typedef itk::RGBPixel< PixelComponentType > PixelType;
+  using PixelType = itk::RGBPixel< PixelComponentType >;
 
   const   unsigned int        Dimension = 3;
```
Comment (dzenanz) [inline SoftwareGuide/Cover/Source/ImageReadExtractWriteRGB.cxx]: Alignment

File: SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex [hunk 8]
```diff
@@ -2339,30 +2339,6 @@ \subsection{Alignment}
 \end{minted}
 \normalsize
 
-Consecutive typedefs should be aligned. If white spaces need to be added to a
```
Comment (dzenanz) [inline SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex]: Awesome!

File: SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex [hunk 9]
```diff
@@ -2781,8 +2757,8 @@ \subsection{Empty Lines}
   //
   const unsigned int Dimension = 2;
```
Comment (dzenanz) [inline SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex]: Dimension already looking good :smile:

**Review by dzenanz — APPROVED** [review]
Great work Hans!

**Review by dzenanz — APPROVED** [review]
Looks good!

**Review by thewtex — APPROVED** [review]
Thanks, Hans!

As a side note, I noticed that we state that `typedef`'s should not be used with `enum`'s -- so, if they are present in ITK now, they were not intended.

File: SoftwareGuide/Cover/Source/VWHistogramHSV.cxx [hunk 10]
```diff
@@ -66,19 +66,19 @@ int main(int argc, char * argv[] )
     return -1;
   }
 
-  typedef unsigned char                        PixelComponentType;
-  typedef itk::RGBPixel<PixelComponentType>    InputPixelType;
-  typedef unsigned short                       OutputPixelType;
+  using PixelComponentType =  unsigned char                        ;
```
Comment (jhlegarreta) [inline SoftwareGuide/Cover/Source/VWHistogramHSV.cxx]: May be too nitpick, but I would **eliminate the white spaces before the semicolon**.
I cannot remember whether this is allowed by the `KWStyle` checker.
There are multiple instances.
Thanks.

**Review by jhlegarreta — CHANGES_REQUESTED** [review]
Thanks for doing this @hjmjohnson

File: SoftwareGuide/Cover/Source/AntialiasFilter.cxx (hunk 2 above)
Comment (hjmjohnson) [reply SoftwareGuide/Cover/Source/AntialiasFilter.cxx]: Done in last whitespace fix.

File: SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx (hunk 4 above)
Comment (hjmjohnson) [reply SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx]: Done in last whitespace fix.

File: SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx (hunk 5 above)
Comment (hjmjohnson) [reply SoftwareGuide/Cover/Source/BinaryThresholdFilter.cxx]: Done in last whitespace fix.

File: SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex (hunk 8 above)
Comment (hjmjohnson) [reply SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex]: 🥇   This is my primary motivation for working so hard on this!

**Comment by hjmjohnson** [comment]
@jhlegarreta If you approve these changes, please go ahead and merge.

**Review by jhlegarreta — APPROVED** [review]

**Comment by jhlegarreta** [comment]
Thanks @hjmjohnson and the rest of you for doing the huge effort required at all levels by the C++11 transition !

**Pull request merged**